res/controllers/hid-controller-api.d.ts

```diff
@@ -0,0 +1,47 @@
+
+declare class HidControllerJSProxy {
+    /** Sends HID OutputReport with hard coded ReportID 0 to HID device
+     *        This function only works with HID devices, which don't use ReportIDs
+     * @param dataList Data to send as list of bytes
+     * @param length This optional argument is no longer evaluated, and only here for backwards compatibility
+    */
+    public send(dataList:number[], length:number=0):void;
+
+    /** Sends HID OutputReport to HID device
+     * @param dataList Data to send as list of bytes
+     * @param length Unused but mandatory argument
+     * @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use ReportIDs
+     * @param resendUnchangedReport If set, the report will also be send, if the data are unchanged since last sending
+    */
+    public send(dataList:number[], length:number, reportID:number, resendUnchangedReport:boolean = false):void;
+
+    /** Sends an OutputReport to HID device
+     *  @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use ReportIDs
+     *  @param dataArray Data to send as byte array (Javascript type Uint8Array)
+     *  @param resendUnchangedReport If set, the report will also be send, if the data are unchanged since last sending
```

Comment:
Maybe this here: https://embeddedguruji.blogspot.com/2019/04/learning-usb-hid-in-linux-part-1-basics.html

Reply:
Mhmm, I don't think I can review this properly because my understanding of HID seems to be wrong...

Reply:
I'm pretty sure, that the documetation of the HID part is correct, since I digged very deep into it and implemented most of these functions myself.
While I'm not so familar with MIDI, Engine and Console.